### PR TITLE
fix: missing dependency for clr-tree-node

### PR DESCRIPTION
**Pull request by bdryanovski**
## PR Checklist

Please check if your PR fulfills the following requirements:

* [ ] Tests for the changes have been added (for bug fixes / features)
* [x] Docs have been added / updated (for bug fixes / features)
* [ ] If applicable, have a visual design approval

## PR Type

Close: #3947

* [x] Bugfix
* [ ] Feature
* [ ] Code style update (formatting, local variables)
* [ ] Refactoring (no functional changes, no api changes)
* [ ] Build related changes
* [ ] CI related changes
* [x] Documentation content changes
* [ ] clarity.design website / infrastructure changes
* [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #3947 

## Does this PR introduce a breaking change?

* [ ] Yes
* [x] No

### CODE REVIEW

**Comment by bdryanovski** [comment]
Not sure if this is the best solution for that issue.

I think that the service must be provided for `ClrTreeViewModule` only.  But this leads to a lot of failing tests - not sure are the tests wrong or me?

```bash
FAILED TESTS:
  Tree View
    Components
      ClrTreeNode Component
        A11y and Focus Management
          ✖ focuses node content container if focus is requested
            HeadlessChrome 78.0.3904 (Mac OS X 10.14.6)
          Error: Expected <body style="cursor: auto;">...</body> to be <div class="clr-tree-node-content-container" role="treeitem" tabindex="-1" aria-expanded="false" aria-selected="false">...</div>. Tip: To check for deep equality, use .toEqual() instead of .toBe().
              at <Jasmine>
              at UserContext.<anonymous> (data/tree-view/tree-node.spec.ts:381:40)
              at ZoneDelegate.push.../../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke (<project-path>/node_modules/zone.js/dist/zone.js:391:1)
              at ProxyZoneSpec.push.../../node_modules/zone.js/dist/zone-testing.js.ProxyZoneSpec.onInvoke (<project-path>/node_modules/zone.js/dist/zone-testing.js:308:1)
          ✖ assigns tabindex of 0 to content container if focus is requested
            HeadlessChrome 78.0.3904 (Mac OS X 10.14.6)
          Error: Expected -1 to be 0.
              at <Jasmine>
              at UserContext.<anonymous> (data/tree-view/tree-node.spec.ts:389:64)
              at ZoneDelegate.push.../../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke (<project-path>/node_modules/zone.js/dist/zone.js:391:1)
              at ProxyZoneSpec.push.../../node_modules/zone.js/dist/zone-testing.js.ProxyZoneSpec.onInvoke (<project-path>/node_modules/zone.js/dist/zone-testing.js:308:1)
          ✖ calls focusManager.focusFirstVisibleNode on Home key
            HeadlessChrome 78.0.3904 (Mac OS X 10.14.6)
          Error: Expected spy focusFirstVisibleNode to have been called.
              at <Jasmine>
              at UserContext.<anonymous> (data/tree-view/tree-node.spec.ts:423:52)
              at ZoneDelegate.push.../../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke (<project-path>/node_modules/zone.js/dist/zone.js:391:1)
              at ProxyZoneSpec.push.../../node_modules/zone.js/dist/zone-testing.js.ProxyZoneSpec.onInvoke (<project-path>/node_modules/zone.js/dist/zone-testing.js:308:1)
          ✖ calls focusManager.focusLastVisibleNode on End key
            HeadlessChrome 78.0.3904 (Mac OS X 10.14.6)
          Error: Expected spy focusLastVisibleNode to have been called.
              at <Jasmine>
              at UserContext.<anonymous> (data/tree-view/tree-node.spec.ts:429:51)
              at ZoneDelegate.push.../../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke (<project-path>/node_modules/zone.js/dist/zone.js:391:1)
              at ProxyZoneSpec.push.../../node_modules/zone.js/dist/zone-testing.js.ProxyZoneSpec.onInvoke (<project-path>/node_modules/zone.js/dist/zone-testing.js:308:1)
          ✖ calls focusManager.focusNodeAbove on ArrowUp key
            HeadlessChrome 78.0.3904 (Mac OS X 10.14.6)
          Error: Expected spy focusNodeAbove to have been called with [ DeclarativeTreeNodeModel({ selected: BehaviorSubject({ _isScalar: false, observers: [ FilterSubscriber({ closed: false, _parentOrParents: Subscriber({ closed: false, _parentOrParents: null, _subscriptions: [ <circular reference: Object> ], syncErrorValue: null, syncErrorThrown: false, syncErrorThrowable: true, isStopped: false, destination: SafeSubscriber({ closed: false, _parentOrParents: null, _subscriptions: null, syncErrorValue: null, syncErrorThrown: false, syncErrorThrowable: false, isStopped: false, destination: Object({ closed: true, next: Function, error: Function, complete: Function }), _parentSubscriber: <circular reference: Object>, _context: <circular reference: Object>, _next: Function, _error: undefined, _complete: undefined }) }), _subscriptions: [ SubjectSubscription({ closed: false, _parentOrParents: <circular reference: Object>, _subscriptions: null, subject: <circular reference: Object>, subscriber: <circular reference: Object> }) ], syncErrorValue: null, sync ... but it was never called.
              at <Jasmine>
              at UserContext.<anonymous> (data/tree-view/tree-node.spec.ts:435:45)
              at ZoneDelegate.push.../../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke (<project-path>/node_modules/zone.js/dist/zone.js:391:1)
              at ProxyZoneSpec.push.../../node_modules/zone.js/dist/zone-testing.js.ProxyZoneSpec.onInvoke (<project-path>/node_modules/zone.js/dist/zone-testing.js:308:1)
          ✖ calls focusManager.focusNodeBelow on ArrowDown key
            HeadlessChrome 78.0.3904 (Mac OS X 10.14.6)
          Error: Expected spy focusNodeBelow to have been called with [ DeclarativeTreeNodeModel({ selected: BehaviorSubject({ _isScalar: false, observers: [ FilterSubscriber({ closed: false, _parentOrParents: Subscriber({ closed: false, _parentOrParents: null, _subscriptions: [ <circular reference: Object> ], syncErrorValue: null, syncErrorThrown: false, syncErrorThrowable: true, isStopped: false, destination: SafeSubscriber({ closed: false, _parentOrParents: null, _subscriptions: null, syncErrorValue: null, syncErrorThrown: false, syncErrorThrowable: false, isStopped: false, destination: Object({ closed: true, next: Function, error: Function, complete: Function }), _parentSubscriber: <circular reference: Object>, _context: <circular reference: Object>, _next: Function, _error: undefined, _complete: undefined }) }), _subscriptions: [ SubjectSubscription({ closed: false, _parentOrParents: <circular reference: Object>, _subscriptions: null, subject: <circular reference: Object>, subscriber: <circular reference: Object> }) ], syncErrorValue: null, sync ... but it was never called.
              at <Jasmine>
              at UserContext.<anonymous> (data/tree-view/tree-node.spec.ts:441:45)
              at ZoneDelegate.push.../../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke (<project-path>/node_modules/zone.js/dist/zone.js:391:1)
              at ProxyZoneSpec.push.../../node_modules/zone.js/dist/zone-testing.js.ProxyZoneSpec.onInvoke (<project-path>/node_modules/zone.js/dist/zone-testing.js:308:1)
          ✖ calls focusManager.focusNodeBelow if node is already expanded on ArrowRight key
            HeadlessChrome 78.0.3904 (Mac OS X 10.14.6)
          Error: Expected spy focusNodeBelow to have been called with [ DeclarativeTreeNodeModel({ selected: BehaviorSubject({ _isScalar: false, observers: [ FilterSubscriber({ closed: false, _parentOrParents: Subscriber({ closed: false, _parentOrParents: null, _subscriptions: [ <circular reference: Object> ], syncErrorValue: null, syncErrorThrown: false, syncErrorThrowable: true, isStopped: false, destination: SafeSubscriber({ closed: false, _parentOrParents: null, _subscriptions: null, syncErrorValue: null, syncErrorThrown: false, syncErrorThrowable: false, isStopped: false, destination: Object({ closed: true, next: Function, error: Function, complete: Function }), _parentSubscriber: <circular reference: Object>, _context: <circular reference: Object>, _next: Function, _error: undefined, _complete: undefined }) }), _subscriptions: [ SubjectSubscription({ closed: false, _parentOrParents: <circular reference: Object>, _subscriptions: null, subject: <circular reference: Object>, subscriber: <circular reference: Object> }) ], syncErrorValue: null, sync ... but it was never called.
              at <Jasmine>
              at UserContext.<anonymous> (data/tree-view/tree-node.spec.ts:467:45)
              at ZoneDelegate.push.../../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke (<project-path>/node_modules/zone.js/dist/zone.js:391:1)
              at ProxyZoneSpec.push.../../node_modules/zone.js/dist/zone-testing.js.ProxyZoneSpec.onInvoke (<project-path>/node_modules/zone.js/dist/zone-testing.js:308:1)
          ✖ calls focusManager.focusParent if node is already collapsed or not expandable on ArrowLeft key
            HeadlessChrome 78.0.3904 (Mac OS X 10.14.6)
          Error: Expected spy focusParent to have been called with [ DeclarativeTreeNodeModel({ selected: BehaviorSubject({ _isScalar: false, observers: [ FilterSubscriber({ closed: false, _parentOrParents: Subscriber({ closed: false, _parentOrParents: null, _subscriptions: [ <circular reference: Object> ], syncErrorValue: null, syncErrorThrown: false, syncErrorThrowable: true, isStopped: false, destination: SafeSubscriber({ closed: false, _parentOrParents: null, _subscriptions: null, syncErrorValue: null, syncErrorThrown: false, syncErrorThrowable: false, isStopped: false, destination: Object({ closed: true, next: Function, error: Function, complete: Function }), _parentSubscriber: <circular reference: Object>, _context: <circular reference: Object>, _next: Function, _error: undefined, _complete: undefined }) }), _subscriptions: [ SubjectSubscription({ closed: false, _parentOrParents: <circular reference: Object>, _subscriptions: null, subject: <circular reference: Object>, subscriber: <circular reference: Object> }) ], syncErrorValue: null, sync ... but it was never called.
              at <Jasmine>
              at UserContext.<anonymous> (data/tree-view/tree-node.spec.ts:480:42)
              at ZoneDelegate.push.../../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke (<project-path>/node_modules/zone.js/dist/zone.js:391:1)
              at ProxyZoneSpec.push.../../node_modules/zone.js/dist/zone-testing.js.ProxyZoneSpec.onInvoke (<project-path>/node_modules/zone.js/dist/zone-testing.js:308:1)
TOTAL: 8 FAILED, 2984 SUCCESS
```

@Shijir

**Comment by bdryanovski** [comment]
This is just masking the problem on our side.

If two components have a dependency like `clrTreeNode` require to be always inside the `clrTree` we need to provide an error to notify the user that some of the components require parent component to work. 

There is no such requirement mention in the documentation. There are even a lot of examples of just having the `clr-tree-node` so this type of error will be something that many people will get in there implementation.  

Another option could be to make this dependency optional for the `clr-tree-node` but this will make the focus management not working if not provided.

We could update the documentation for Tree View and provide examples that all `clr-tree-node` have to be part of at least one `clr-tree` to work.

I'm not sure that we are the first one to use only the `clr-tree-node` syntax out there.

**Comment by Shijir** [comment]
As Jeremy mentioned, this is more of a website demo and documentation update issue. I've updated all demo code examples in the Tree View's page.

**Comment by bdryanovski** [comment]
OK updated the documentation and the badges demo page.

File: src/website/src/app/documentation/demos/tree-view/tree-view.demo.html [hunk 1]
```diff
@@ -300,6 +300,21 @@ <h3 id="basic-tree">Basic Tree</h3>
...
+            <div class="alert alert-info has-margin-top-b2">
+                <div class="alert-items">
+                    <div class="alert-item static">
+                        <div class="alert-icon-wrapper">
+                            <clr-icon class="alert-icon" shape="info-circle"></clr-icon>
+                        </div>
+                        <div class="alert-text">
+                            Please note that every tree require to have root node to work properly,
+                            this is done by having <code class="clr-code">clr-tree</code> as root wrapper, you
+                            could check the example below.
```
Comment (Jinnie) [inline src/website/src/app/documentation/demos/tree-view/tree-view.demo.html]: nit: require -> requires
I'd also change "you could check" -> "please check", or even drop the entire last sub-sentence as it hijacks the goal of the example.

**Pull request merged**